cloud/misc/ovirt_storage.py

```diff
@@ -0,0 +1,432 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# (c) 2016, Ryan Groten <rgroten@gmail.com>
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: ovirt_storage
+author: "Ryan Groten (@rgroten)"
+short_description: "Create/Delete/Manage VM Disks in oVirt/RHEV"
+description:
+    - Module for managing disks in oVirt/RHEV. Provides the ability to create, attach/detach, or delete both internal (pool) disks to VMs.
+version_added: "2.2"
+requirements:
+    - "ovirt-engine-python-sdk"
+options:
+    state:
+        description:
+            - C(present) will create a new disk and attach it to a VM.
+            - C(absent) will delete a disk attached to a VM.
+            - C(attach) will attach a disk to a VM.
+            - C(detach) will detach a disk from a VM.
+        choices: [ 'present', 'absent', 'attached', 'detached' ]
+        default: present
+    user:
+        description:
+            - User with access to perform disk operations in oVirt.
+        required: true
+    password:
+        description:
+            - Password for I(user).
+        required: true
+    url:
+        description:
+            - URL to oVirt API (U(https://<ovirt-engine>/api)).
```

Comment:
The path of the oVirt API should be `/ovirt-engine/api`. The old `/api` has been deprecated and will be removed in version 4.1 of oVirt.